test_runner/performance/test_physical_replication.py

```diff
@@ -0,0 +1,232 @@
+            cur_replica.append(conn.cursor())
+
+        # Sync replicas
+        for i in range(num_replicas):
+            measure_replication_lag(cur_master, cur_replica[i])
+
+        master_pgbench = pg_bin.run_nonblocking(
+            [
+                "pgbench",
+                "-c10",
+                "-T1000",
+                "-Mprepared",
+                "--log",
+                f"--log-prefix={test_output_dir}/pgbench_master",
+                f"--aggregate-interval={configuration_test_time_sec}",
+                master_connstr,
+            ]
+        )
+        replica_pgbench: List[Optional[subprocess.Popen[Any]]] = [None for i in range(num_replicas)]
+
+        for iconfig in range((1 << num_replicas) - 1, -1, -1):
+
+            def replica_enabled(ireplica, iconfig=iconfig):
+                return bool((iconfig >> 1) & 1)
+
+            # Change configuration
+            for ireplica in range(num_replicas):
+                if replica_enabled(ireplica) and replica_pgbench[ireplica] is None:
+                    replica_pgbench[ireplica] = pg_bin.run_nonblocking(
+                        [
+                            "pgbench",
+                            "-c10",
+                            "-S",
+                            "-T1000",
+                            "--log",
+                            f"--log-prefix={test_output_dir}/pgbench_replica_{ireplica}",
+                            f"--aggregate-interval={configuration_test_time_sec}",
+                            replica_connstr[ireplica],
+                        ]
+                    )
+                elif not replica_enabled(ireplica) and replica_pgbench[ireplica] is not None:
+                    pgb = replica_pgbench[ireplica]
+                    assert pgb is not None
+                    pgb.terminate()
+                    pgb.wait()
+                    replica_pgbench[ireplica] = None
+                    conn_replica[ireplica] = None
+                    cur_replica[ireplica] = None
+
+                    neon_suspend_endpoint(
+                        neon_api_key,
+                        neon_api_base_url,
+                        project_id,
+                        replicas[ireplica]["endpoint"]["id"],
+                    )
+                    neon_wait_for_operation_to_finish(neon_api_key, neon_api_base_url, project_id)
+            time.sleep(configuration_test_time_sec)
+
+            for ireplica in range(num_replicas):
+                conn = conn_replica[ireplica]
+                if conn is None:
+                    conn = psycopg2.connect(replica_connstr[i])
+                    conn_replica[ireplica] = conn
+                    cur_replica[ireplica] = conn.cursor()
+
+                lag = measure_replication_lag(cur_master, cur_replica[ireplica])
+                log.info(
+                    f"Replica {ireplica} lagging behind master by {lag} seconds after configuration {iconfig:>b}"
+                )
```

Comment:
We should use `zenbenchmark` fixture to store results in a database, here's an example:
https://github.com/neondatabase/neon/blob/91dd99038e8e85a29d028fe59e9abd1cc978e415/test_runner/performance/test_gc_feedback.py#L79-L83

Reply:
Made it record with zenbenchmark, please take a look to see if what I did makes sense - I haven't used this system much before